pkg/controller/controller.go

```diff
@@ -33,8 +33,12 @@ const (
 
 // MachineApproverReconciler reconciles a machine-approver  object
 type CertificateApprover struct {
-	client.Client
-	RestCfg  *rest.Config
+	NodeClient  client.Client
+	NodeRestCfg *rest.Config
+
+	MachineClient  client.Client
+	MachineRestCfg *rest.Config
```

Comment:
why do we need `MachineRestCfg`? we already have the client

Reply:
it is used for API discovery, following austoscaler pattern here too https://github.com/kubernetes/autoscaler/blob/8dc1afb9a2fd188374620ee0d42358243e087655/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go#L331